catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/SecureHadoopCatalogOperations.java

```diff
@@ -50,28 +61,33 @@
 import org.apache.gravitino.file.FilesetChange;
 import org.apache.gravitino.meta.FilesetEntity;
 import org.apache.gravitino.meta.SchemaEntity;
+import org.apache.gravitino.utils.NameIdentifierUtil;
 import org.apache.gravitino.utils.PrincipalUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @SuppressWarnings("removal")
 public class SecureHadoopCatalogOperations
-    implements CatalogOperations, SupportsSchemas, FilesetCatalog {
+    implements CatalogOperations, SupportsSchemas, FilesetCatalog, SupportsCredentialOperations {
 
   public static final Logger LOG = LoggerFactory.getLogger(SecureHadoopCatalogOperations.class);
 
   private final HadoopCatalogOperations hadoopCatalogOperations;
+  private CatalogCredentialManager catalogCredentialManager;
```

Comment:
Can we achieve this credential operations in the core module, like in operation dispatcher? It's a bit tricky to mix this with the catalog, WDYT?

Reply:
You mean like tag operation?

Reply:
Yes, my feeling is that the credential may be orthogonal to the catalog implementation, we don't have to embed the code to each catalog implementation.

Reply:
Let me think more, what's different for Credential operations is the classloader, it should load the credential provider in catalog classloader.